tests/test_queries.py

```diff
@@ -367,7 +367,94 @@ def test_query_api(client):
+
+    json = qmcall(
+        c, {"q": "hvenær fer næsta flug frá keflavík", "voice": True}, "Flights"
+    )
+    assert re.search(departure_pattern, json["answer"])
+    json = qmcall(
+        c, {"q": "hvenær flýgur næsta flug frá keflavík", "voice": True}, "Flights"
+    )
+    assert re.search(departure_pattern, json["answer"])
+    json = qmcall(
+        c,
+        {"q": "hvenær flýgur næsta flug af stað frá keflavík", "voice": True},
+        "Flights",
+    )
+    assert re.search(departure_pattern, json["answer"])
+    json = qmcall(
+        c,
+        {"q": "hver er brottfarartími næsta flugs frá keflavík", "voice": True},
+        "Flights",
+    )
+    assert re.search(departure_pattern, json["answer"])
+    json = qmcall(
+        c,
+        {"q": "hver er brottfarartíminn fyrir næsta flug frá keflavík", "voice": True},
+        "Flights",
+    )
+    assert re.search(departure_pattern, json["answer"])
+
+    json = qmcall(
+        c, {"q": "hvenær lendir næsta flug í keflavík", "voice": True}, "Flights"
+    )
+    assert re.search(arrival_pattern, json["answer"])
+    json = qmcall(
+        c, {"q": "hvenær kemur næsta vél á akureyri", "voice": True}, "Flights"
+    )
+    assert re.search(arrival_pattern, json["answer"]) or re.search(
+        no_matching_flight_pattern, json["answer"]
+    )  # In case no flights to Akureyri
+    json = qmcall(
+        c, {"q": "hvenær mætir næsta vél á vopnafjörð", "voice": True}, "Flights"
+    )
+    assert re.search(arrival_pattern, json["answer"]) or re.search(
+        no_matching_flight_pattern, json["answer"]
+    )  # In case no flights to Vopnafjörður
+    json = qmcall(
+        c, {"q": "hvenær mætir næsta vél til vopnafjarðar", "voice": True}, "Flights"
+    )
+    assert re.search(arrival_pattern, json["answer"]) or re.search(
+        no_matching_flight_pattern, json["answer"]
+    )  # In case no flights to Vopnafjörður
+    json = qmcall(
+        c,
+        {
+            "q": "hver er lendingartími næstu vélar frá reykjavík til vopnafjarðar",
+            "voice": True,
+        },
+        "Flights",
+    )
+    assert re.search(arrival_pattern, json["answer"]) or re.search(
+        no_matching_flight_pattern, json["answer"]
+    )
+    json = qmcall(
+        c,
+        {
+            "q": "hver er lendingartíminn fyrir næsta flug til reykjavíkur frá akureyri",
+            "voice": True,
+        },
+        "Flights",
+    )
+    assert re.search(arrival_pattern, json["answer"]) or re.search(
+        no_matching_flight_pattern, json["answer"]
+    )
+
+    json = qmcall(
+        c,
+        {"q": "hvenær fer næsta flug til blabla frá ekkitil", "voice": True},
+        "Flights",
+    )
+    assert re.search(no_matching_flight_pattern, json["answer"])
+    json = qmcall(
+        c,
+        {"q": "hvenær fer næsta flug frá ekkitil til blablab", "voice": True},
+        "Flights",
+    )
+    assert re.search(no_matching_flight_pattern, json["answer"])
```

Comment:
I hesitate to say it, but this is a lot of tests vis-a-vis the tests for the other query modules in test_queries. Our continuous integration testing pipeline is taking long enough as it is, and all these HTTP queries are quite expensive. Are they all necessary? @vthorsteinsson What's our policy on this? Do we want full HTTP comm testing on all the query modules or should we move over to testing the queries directly via code rather than through the API? Seems like the way to go to me.

Reply:
I see your point, but on the other hand there can almost never be too much testing! I hazard a guess that the HTTP overhead is pretty negligible compared to the actual query processing, so I say we let this stay as-is (and not be too reluctant to add more tests of this nature to other queries).

Reply:
OK, I'll extend the tests, have been reluctant to do so hitherto due to basic performance issues.